### PR TITLE
sync with controller, remove the dockerclient dependency

### PR DESCRIPTION
**Pull request by pohanhuangtw**
### Summary

- Sync with controller
- Remove the dockerclient dependency

### CODE REVIEW

File: cvetools/image.go [hunk 1]
```diff
@@ -112,7 +114,7 @@ func (s *ScanTools) LoadLocalImage(ctx context.Context, repository, tag, imgPath
 	histories, err := rt.GetImageHistory(imageName)
 	if err != nil {
 		log.WithFields(log.Fields{"error": err}).Error("Failed to get local image history")
-		if err == dockerclient.ErrImageNotFound {
+		if err == ErrImageNotFound {
```
Comment (jayhuang-suse) [inline cvetools/image.go:117]: Please refer below line to apply the error check.

https://github.com/moby/moby/blob/ca0910c1b903c8dd437dd24fd9a68d92dee17a2c/client/image_inspect_test.go#L36C18-L36C55

**Review by jayhuang-suse — CHANGES_REQUESTED** [review]

**Comment by jayhuang-suse** [comment]
FYI, it is synced to neuvector repo, not only its controller.

File: cvetools/image.go (hunk 1 above)
Comment (pohanhuangtw) [reply cvetools/image.go]: Thanks for reminding.

**Review by jayhuang-suse — CHANGES_REQUESTED** [review]

File: cvetools/image.go [hunk 2]
```diff
@@ -14,13 +14,13 @@ import (
 	"strings"
 	"time"
 
+	"github.com/containerd/errdefs"
```
[the remark this answers is not in the document]
Comment (pohanhuangtw) [reply cvetools/image.go]: ok

**Review by jayhuang-suse — APPROVED** [review]
LGTM

**Pull request merged**